packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.tsx

```diff
@@ -93,22 +85,30 @@ export function AutoSaveField<T = string>(props: Props<T>) {
    * use Field around input to pass the error message
    * use InlineToast.tsx to show the save message
    */
+  const styles = useStyles2(getStyles);
+
   return (
     <>
       <Field
         {...restProps}
-        loading={isLoading}
+        loading={loading || undefined}
         invalid={isInvalid}
         disabled={disabled}
         error={error || (fieldState.showError && saveErrorMessage)}
         ref={fieldRef}
+        className={styles.widthFitContent}
       >
         {React.cloneElement(
           children((newValue) => {
             lodashDebounce(newValue);
           })
         )}
       </Field>
+      {isLoading && (
```

Comment:
Should this be only on `fieldState.loading`? Is that the one that happens when it is saving?

Reply:
Yes, you're right 😅